pkg/agent/cniserver/server.go

```diff
@@ -556,6 +559,15 @@ func (s *CNIServer) cmdDel(_ context.Context, cniConfig *CNIConfig) (*cnipb.CniC
 	}
 	klog.InfoS("Deleted interfaces for container", "container", cniConfig.ContainerId)
 
+	// Check if any SR-IOV device is still attached before removing primary interfaces.
+	if s.enableSecondaryNetwork && s.vfDeviceChecker != nil {
```

Comment:
based on usage, it doesn't feel like the `enableSecondaryNetwork` is actually needed. Can't you just use `s.vfDeviceChecker != nil`, as `s.vfDeviceChecker` should be `nil` when the SecondaryNetwork feature is disabled?